tests/Examples/UnionExample.hs

```diff
@@ -17,13 +17,13 @@ type O2 = Object "O2" '[] '[Field "o2" Text]
 type T = Union "U" '[O1, O2]
 
 o1 :: Handler IO O1
-o1 = pure ((pure "hello from O1") :<> ())
+o1 = pure (pure "hello from O1")
 
 o2 :: Handler IO O2
-o2 = pure ((pure "hello from O2") :<> ())
+o2 = pure (pure "hello from O2")
 
 tHandler :: Handler IO T
-tHandler = o1 :<|> o2 :<|> ()
+tHandler = o1 :<|> o2
```

Comment:
These operators (`:<|>` and `:<>`) are **not** associative! That's the reason we can't have the `fieldList` function. They construct a structure that's isomorphic to a right-nested tuple. Any re-nesting of that tuple will result in a different type.

It might possible to lift that restriction at some point but it's beyond my capabilities ATM.